ibis/backends/flink/__init__.py

```diff
@@ -677,9 +677,20 @@ def _read_file(
             If `schema` is None.
         """
         if schema is None:
-            raise ValueError(
-                f"`schema` must be explicitly provided when calling `read_{file_type}`"
-            )
+            if file_type == "parquet":
+                import pyarrow as pa
+
+                try:
+                    pyarrow_schema = pa.parquet.read_schema(path)
+                except FileNotFoundError:
+                    raise ValueError(f"No file found at {path}")
```

Comment:
This should handle single files, directories, and lists of files (it won't handle globs though):

```suggestion
                import pyarrow.dataset as ds

                pyarrow_schema = ds.datasets(path, format="parquet").schema
```

I also think it's fine to let the original pyarrow error bubble up.

Reply:
Thanks. I made the change you suggested, however I have a few comments and concerns around this:

* **Supporting directories**: My concern here is the partitioning used in the directory. Flink's file system [supports the standard hive format](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/filesystem/#partition-files) while Arrow seems to support [a richer set of partitioning strategies](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.dataset.html). 

* **List of paths**: Even though `ds.dataset()` is capable of constructing the schema for a list of paths, I am not sure (1) if Flink's FileSystem connector supports this, (2) if it supports, how to "activate" it. There is high-level information in [Flink documentation](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/filesystem/#file-source) but not immediately clear in addressing (1) and (2).
I will look more into this.

* The comment for `read_parquet()` (e.g., in `ibis/backends/base/__init__.py`) mention the support for only files:
```
Register a parquet file as a table in the current backend.
```
Should we modify this to explain the fact that a single or multiple files can be registered depending on the specific backend implementation. This comment was confusing to me while working on this PR :)

**Summary**: I will mark this PR draft and spend some time reading on the concerns raised above.

Reply:
> That is why I will exclude this unless you have a different suggestion.

All I meant by my suggestion to use `pyarrow.dataset` was that it supported a wider variety of input formats than just a single file. If flink doesn't support anything but a single file here (instead of a directory, glob, or list of files) then we can drop this for now.

What you have here now though is probably fine - if flink doesn't support an input an error will still be raised by flink. The only fail case is if the pyarrow datasets infers a different schema than what flink would infer due to partitioning etc... If this is something that could happen, reverting to explicitly using `pyarrow.parquet.read_metadata(path).schema.to_arrow_schema()` is probably better. This will let pyarrow determine if something is a single path (and will error appropriately otherwise) while still retrieving the schema from the file metadata. We can always add support for a directory/glob/list of files later as needed.

Reply:
Thanks. I would prefer "more predictable" over additional support with silent failures. I switched to `pyarrow.parquet.read_metadata(path).schema.to_arrow_schema()`. Added a note in `read_parquet()` to explain why we only support single files for now.